src/iop/colorequal.c

```diff
@@ -0,0 +1,1384 @@
+
+#ifdef _OPENMP
+#pragma omp parallel for simd default(none) \
+  dt_omp_firstprivate(ds_pixels, ds_U, ds_V, covariance)  \
+  schedule(simd:static) aligned(ds_U, ds_V, covariance: 64)
+#endif
+  for(size_t k = 0; k < ds_pixels; k++)
+  {
+    // corr(U, U)
+    covariance[4 * k + 0] = ds_U[k] * ds_U[k];
+    // corr(U, V)
+    covariance[4 * k + 1] = covariance[4 * k + 2] = ds_U[k] * ds_V[k];
+    // corr(V, V)
+    covariance[4 * k + 3] = ds_V[k] * ds_V[k];
+  }
+
+  // We actually compute 3 guided filters here :
+  // saturation = corrections[0] = a_sat * UV + b_sat
+  // hue        = corrections[1] = a_hue * UV + b_hue
+  // brightness = corrections[2] = a_bright * UV + b_bright
+  // Generate here the 6 correlations between saturation, hue, brightness
+  // and U, V.
+  float *const restrict sat_uv = dt_alloc_align_float(ds_pixels * 2);
+  float *const restrict hue_uv = dt_alloc_align_float(ds_pixels * 2);
+  float *const restrict bri_uv = dt_alloc_align_float(ds_pixels * 2);
+
+#ifdef _OPENMP
+#pragma omp parallel for simd default(none) \
+  dt_omp_firstprivate(ds_pixels, sat_uv, hue_uv, bri_uv, ds_U, ds_V, ds_corrections, stdout)  \
+  schedule(simd:static) aligned(sat_uv, hue_uv, bri_uv, ds_U, ds_V, ds_corrections: 64)
+#endif
+  for(size_t k = 0; k < ds_pixels; k++)
+  {
+    // corr(U, saturation)
+    sat_uv[k * 2 + 0] = ds_corrections[k * 4 + 0] * ds_U[k];
+    // corr(V, saturation)
+    sat_uv[k * 2 + 1] = ds_corrections[k * 4 + 0] * ds_V[k];
+
+    //fprintf(stdout, "sat_uv : %f - %f\n", sat_uv[k * 2 + 0], sat_uv[k * 2 + 1]);
+
+    // corr(U, hue)
+    hue_uv[k * 2 + 0] = ds_corrections[k * 4 + 1] * ds_U[k];
+    // corr(V, hue)
+    hue_uv[k * 2 + 1] = ds_corrections[k * 4 + 1] * ds_V[k];
+
+    // corr(U, brightness)
+    bri_uv[k * 2 + 0] = ds_corrections[k * 4 + 2] * ds_U[k];
+    // corr(V, brightness)
+    bri_uv[k * 2 + 1] = ds_corrections[k * 4 + 2] * ds_V[k];
+  }
+
+  // Compute the local averages of everything over the window size
+  // We use a gaussian blur as a weighted local average because it's a radial function
+  // so it will not favour vertical and horizontal edges over diagonal ones
+  // as the by-the-book box blur (unweighted local average) would.
+
+  // We use unbounded signals, so don't care for the internal value clipping
+  float max[4] = { 1000.f };
+  float min[4] = { -1000.f };
+
+  dt_gaussian_t *gauss_1c = dt_gaussian_init(ds_width, ds_height, 1, max, min, ds_sigma, 0);
+  dt_gaussian_blur(gauss_1c, ds_U, ds_U);
+  dt_gaussian_blur(gauss_1c, ds_V, ds_V);
+
+  dt_gaussian_t *gauss_2c = dt_gaussian_init(ds_width, ds_height, 2, max, min, ds_sigma, 0);
+  dt_gaussian_blur(gauss_2c, sat_uv, sat_uv);
+  dt_gaussian_blur(gauss_2c, hue_uv, hue_uv);
+  dt_gaussian_blur(gauss_2c, bri_uv, bri_uv);
+
+  dt_gaussian_t *gauss_4c = dt_gaussian_init(ds_width, ds_height, 4, max, min, ds_sigma, 0);
+  dt_gaussian_blur(gauss_4c, ds_corrections, ds_corrections);
```

Comment:
It seems to me that the guided filter correctly processes only the correction at index 0. You can try it by swapping the indexes in process() as commented below. If you put e.g. brightness at index 0, it suddenly starts working.

Commenting out this gaussian blur line makes all the channels work at the same time. I didn't yet find any root cause for this but will continue looking (probably later this week).